src/webui/www/private/newcategory.html

```diff
@@ -7,50 +7,108 @@
     <link rel="stylesheet" href="css/style.css" type="text/css" />
     <script src="scripts/lib/mootools-1.2-core-yc.js"></script>
     <script src="scripts/lib/mootools-1.2-more.js"></script>
+    <script src="scripts/misc.js"></script>
     <script>
+        console.log("newcategory.html");
         var newCategoryKeyboardEvents = new Keyboard({
             defaultEventType: 'keydown',
             events: {
                 'enter': function(event) {
-                    $('newCategoryButton').click();
+                    $('categoryNameButton').click();
                     event.preventDefault();
                 }
             }
         });
         newCategoryKeyboardEvents.activate();
 
         window.addEvent('domready', function() {
-            $('newCategory').focus();
-            $('newCategoryButton').addEvent('click', function(e) {
-                new Event(e).stop();
-                // check field
-                var categoryName = $('newCategory').value.trim();
-                if (categoryName == null || categoryName == "")
-                    return false;
-                if (categoryName.match("^([^\\\\\\/]|[^\\\\\\/]([^\\\\\\/]|\\/(?=[^\\/]))*[^\\\\\\/])$") === null) {
-                    alert("QBT_TR(Invalid category name:\nPlease do not use any special characters in the category name.)QBT_TR[CONTEXT=HttpServer]");
+            var uriAction = safeTrim(new URI().getData('action'));
```

Comment:
Maybe it is possible to write it as: `String(new URI().getData('action')).trim();` instead of adding a new function?

Reply:
`String(null)` returns `"null"` and `String(undefined)` returns `"undefined"` in JavaScript.